### PR TITLE
undo block remove

### PR DESCRIPTION
**Pull request by filipesilva**
- feat: build-paste-op returns a flat list of atomic ops
- feat: group block save at the end of paste-op
- fix: don't return repr with uid for title or open true
- feat: support :block/remove in undo

### CODE REVIEW

**Review by neotyk — COMMENTED** [review]
I'd like to see more tests around internal representation generation.

File: src/cljc/athens/common_db.cljc [hunk 1]
```diff
@@ -432,19 +432,28 @@
...
 (defn get-internal-representation
   "Returns internal representation for eid in db."
   [db eid]
-  (let [remove-ks [:block/order]
-        rename-ks {:block/open :block/open?
-                   :node/title :page/title}]
+  (let [rename-ks            {:block/open :block/open?
+                              :node/title :page/title}
+        remove-ks-on-match [[:block/order (constantly true)]
+                            [:block/open? #(:block/open? %)]
```
[the remark this answers is not in the document]
Comment (neotyk) [reply src/cljc/athens/common_db.cljc:448]: it's ok to have fn as second in tuple, it's just that `:block/open?` is the same as `#(:block/open? %)`

File: src/cljc/athens/common_events/resolver/undo.cljc [hunk 2]
```diff
@@ -19,10 +19,28 @@
...
   (let [{:block/keys [uid]}    args
         {:block/keys [string]} (common-db/get-block evt-db [:block/uid uid])]
-    [(graph-ops/build-block-save-op db uid string)]))
+    [(atomic-graph-ops/make-block-save-op uid string)]))
+
+
+(defmethod resolve-atomic-op-to-undo-ops :block/remove
+  [_db evt-db {:op/keys [args]}]
+  (let [{:block/keys [uid]} args
+        {:block/keys [order _refs]
```
[the remark this answers is not in the document]
Comment (neotyk) [reply src/cljc/athens/common_events/resolver/undo.cljc:31]: it's more of an observation than a comment

[the remark this answers is not in the document]
Comment (filipesilva) [reply src/cljc/athens/common_events/resolver/undo.cljc:31]: Rebound to `backrefs`.

File: src/cljc/athens/common_db.cljc [hunk 3]
```diff
@@ -432,19 +432,28 @@
...
+    m))
+
+
 (defn get-internal-representation
   "Returns internal representation for eid in db."
   [db eid]
-  (let [remove-ks [:block/order]
-        rename-ks {:block/open :block/open?
-                   :node/title :page/title}]
+  (let [rename-ks            {:block/open :block/open?
```
[the remark this answers is not in the document]
Comment (filipesilva) [reply src/cljc/athens/common_db.cljc:445]: Added

**Comment by filipesilva** [comment]
@neotyk PTAL, also added get-position, and more tests.

**Review by neotyk — APPROVED** [review]

**Pull request merged**